src/main/java/org/terasology/climateConditions/igloo/IglooRasterizer.java

```diff
@@ -0,0 +1,67 @@
+// Copyright 2020 The Terasology Foundation
+// SPDX-License-Identifier: Apache-2.0
+
+package org.terasology.climateConditions.igloo;
+
+import org.terasology.entitySystem.prefab.Prefab;
+import org.terasology.entitySystem.prefab.PrefabManager;
+import org.terasology.math.ChunkMath;
+import org.terasology.math.Region3i;
+import org.terasology.math.geom.BaseVector3i;
+import org.terasology.math.geom.Vector3i;
+import org.terasology.registry.CoreRegistry;
+import org.terasology.registry.In;
+import org.terasology.structureTemplates.components.SpawnBlockRegionsComponent;
+import org.terasology.world.block.Block;
+import org.terasology.world.chunks.CoreChunk;
+import org.terasology.world.generation.Region;
+import org.terasology.world.generation.WorldRasterizerPlugin;
+import org.terasology.world.generator.plugin.RegisterPlugin;
+import org.terasology.world.generator.plugin.WorldGeneratorPluginLibrary;
+
+import java.util.Map;
+import java.util.Objects;
+
+@RegisterPlugin
+public class IglooRasterizer implements WorldRasterizerPlugin {
+    @In
+    private WorldGeneratorPluginLibrary worldGeneratorPluginLibrary;
+
+    /**
+     * Stores the prefab of the Igloo Structure Template.
+     */
+    Prefab iglooStructure;
+
+    @Override
+    public void initialize() {
+        iglooStructure = Objects.requireNonNull(CoreRegistry.get(PrefabManager.class)).getPrefab("ClimateConditions:Igloo");
```

Comment:
nice, this works really well! 👍 

**Potential Follow-ups:** Pinging @AndyTechGuy and @Sin3point14 for some discussion and ideas on how to generalize this for arbitrary structure templates 🤓 One thing coming to my mind is to get the actual template from the facet (or the objects stored in the facet) instead of hard-coding it. We may then be able to provide a common base class for structure-template world rasterizers for reuse by other modules.

Reply:
I did once keep it in the structure class but I just felt that then the prefab manager would be called every time the structure is placed since the class will be initialized every time..... and at the time I was placing a lot of structures... so since I wasn't really using it anywhere else I thought it be best to call it just once in the rasterizer and for borders I created the extra variable in the structure class ..